lib/rubocop/cop/style/guard_clause.rb

```diff
@@ -201,6 +201,12 @@ def autocorrect(corrector, node, condition, replacement, guard)
         end
         # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
 
+        def heredoc?(argument)
+          return false if argument.nil?
+
+          argument.str_type? && argument.heredoc?
```

Comment:
I think the following patch is required to prevent a regression. Can you confirm and update it?

```diff
diff --git a/lib/rubocop/cop/style/guard_clause.rb b/lib/rubocop/cop/style/guard_clause.rb
index cfa80b912..5b2b7d4e1 100644
--- a/lib/rubocop/cop/style/guard_clause.rb
+++ b/lib/rubocop/cop/style/guard_clause.rb
@@ -204,7 +204,7 @@ module RuboCop
         def heredoc?(argument)
           return false if argument.nil?

-          argument.str_type? && argument.heredoc?
+          argument.respond_to?(:heredoc?) && argument.heredoc?
         end

         def autocorrect_heredoc_argument(corrector, node, heredoc_branch, leave_branch, guard)
diff --git a/spec/rubocop/cop/style/guard_clause_spec.rb b/spec/rubocop/cop/style/guard_clause_spec.rb
index 3f70ddc06..c3743b139 100644
--- a/spec/rubocop/cop/style/guard_clause_spec.rb
+++ b/spec/rubocop/cop/style/guard_clause_spec.rb
@@ -297,6 +297,32 @@ RSpec.describe RuboCop::Cop::Style::GuardClause, :config do
     RUBY
   end

+  it 'registers an offense when using xstr heredoc as an argument of raise in `else` branch' do
+    expect_offense(<<~RUBY)
+      def func
+        unless condition
+        ^^^^^^ Use a guard clause (`raise <<~`MESSAGE` unless condition`) instead of wrapping the code inside a conditional expression.
+          raise <<~`MESSAGE`
+            oops
+          MESSAGE
+        else
+          foo
+        end
+      end
+    RUBY
+
+    # NOTE: Let `Layout/HeredocIndentation`, `Layout/ClosingHeredocIndentation`, and
+    #       `Layout/IndentationConsistency` cops autocorrect inconsistent indentations.
+    expect_correction(<<~RUBY)
+      def func
+        raise <<~`MESSAGE` unless condition
+            oops
+          MESSAGE
+      foo
+      end
+    RUBY
+  end
+
   context 'MinBodyLength: 1' do
     let(:cop_config) { { 'MinBodyLength' => 1 } }
```

Reply:
Looks good! I'll fix it right away.